content/blog/2022-04-26-moving-local-experiments-to-the-cloud-with-iterative-terraform-provider.md

```diff
@@ -0,0 +1,305 @@
+  example,
+  [l+k80](https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#l+k80)
+  stands for a "Large, with (at least) 12 CPU cores, 112 GB of RAM and 2 GPU
+  devices".
+- _workdir_: Specify a directory on your local machine relative to your project
+  folder which you would like to sync with the remote machine. This way you can
+  share your whole project or parts of it with a remote machine. In my example,
+  I am sharing the whole project.
+- _script_ (**required)**: This is where the Iterative Provider's magic happens,
+  i.e. this is where we define a script that should be run on a provisioned
+  machine.
+
+See the
+[resource arguments documentation](https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#argument-reference)
+for a full list.
+
+In the simplest scenario, all we need to do on a new machine to run the training
+`script` is to set up the Python environment with required libraries. If you
+simply want to train your model on a machine with more memory, this may be
+enough. However, if you want your training code to leverage GPUs, the script
+will look a bit different.
+
+## Training with GPU
+
+There are several ways you can leverage GPU devices on a remote machine. You can
+install all the required drivers and dependencies "manually" via a script, you
+can use an existing Docker image or build your own, or, in the case of AWS, you
+can take advantage of an
+[existing pre-configured Deep Learning AMI](https://docs.aws.amazon.com/dlami/latest/devguide/options.html).
+In this tutorial, we'll use an AMI, as it is the quickest way to start using
+GPUs on an AWS EC2 machine. An AMI can be selected using the `image` argument
+using the format `{user}@{owner}:{architecture}:{name}`. See the official
+[AMI documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html)
+to learn more.
+
+```hcl
+...
+resource "iterative_task" "tpi-examples-gpu" {
+  name      = "ami-gpu-example"
+  cloud     = "aws"
+  region    = "us-east-2"
+  machine   = "l+k80"
+  disk_size = "130"
+  # see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html for images
+  image     = "ubuntu@898082745236:x86_64:Deep Learning AMI (Ubuntu 18.04) Version 54.0"
+
+  workdir { input = "." }
+  script = <<-END
+    #!/bin/bash
+    pip3 install -r requirements.txt
+    python3 src/train.py
+  END
+}
+```
+
+The Deep Learning image that you choose may require larger disk size than the
+provided default, so you may need to increase it. The image I picked expects a
+volume of at least 130 GB, and I can configure this in the `disk_size` argument.
+
+You may also notice that my script got a bit shorter. This is because I no
+longer need to install Python, and only need to make sure to have my
+requirements met.
+
+## Ready… Set… Apply!
+
+Whether you want to go with the basic example, or the GPU-enabled training, you
+can run
+
+```
+terraform plan
```

Comment:
Getting an error at this point:

```
robdewit@Robs-MacBook-Pro blog-tpi-bees % terraform plan
╷
│ Error: Unsupported block type
│ 
│   on main.tf line 11, in resource "iterative_task" "tpi-examples-basic":
│   11:     workdir { input = "." }
│ 
│ Blocks of type "workdir" are not expected here.
```

Reply:
As stated on https://github.com/iterative/dvc.org/pull/3087#discussion_r850255708, that was subject to change... and changed. See [this guide](https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/getting-started) for working examples.